vortex-layout/src/scan/mod.rs

```diff
@@ -161,6 +174,26 @@ impl<A: 'static + Send> ScanBuilder<A> {
             .map(|f| simplify_typed(f, layout_reader.dtype()))
             .transpose()?;
 
+        if let Some(((file_stats, filter), struct_dtype)) = self
+            .file_stats
+            .zip(self.filter.as_ref())
+            .zip(layout_reader.dtype().as_struct())
+        {
+            if let Some(predicate) = PruningPredicate::try_new(filter) {
+                if let Some(struct_row) = extract_struct_row(&predicate, &file_stats, struct_dtype)?
+                {
+                    if let Some(result) = predicate
+                        .evaluate(&struct_row)?
```

Comment:
Can you fix the evaluate function to take a trait? Maybe it looks like:

```
StatsSource {
 fn fetch(&self, field_path: &FieldPath, stat: &Stat) -> Option<ArrayRef>;
}
```

That way you don't need to construct the full stats array if we only end up using one field.

Reply:
We already do this, would you prefer this method to live in a impl/trait?